src/components/share/modal/ShareModal.tsx

```diff
@@ -231,9 +259,16 @@ export const ShareModal = <UserType, InvitationType, AccessType>({
   };
 
   const showInvitations =
-    !showSearchUsers && !props.loading && invitations.length > 0;
+    !hideInvitations &&
+    !showSearchUsers &&
+    !props.loading &&
+    invitations.length > 0;
+
+  const showMembers =
+    !hideMembers && !showSearchUsers && !props.loading && members.length > 0;
 
-  const showMembers = !showSearchUsers && !props.loading && members.length > 0;
+  // If we hide invitations and members, we don't show the search anyway.
+  const showSearch = !(hideInvitations && hideMembers);
```

Comment:
Here I have the impression that you are talking about the input (there is a prop in the QuickSearch to hide it just in case). Moreover, hiding everything is complicated because there may be a given child that has nothing to do with this rule. We can talk about it when you are on this point